### PR TITLE
Remove obsolete service message system commands

### PR DESCRIPTION
**Pull request by noplanman**
| ?            |  !
|---           | ---
| Type         | bug / improvement
| BC Break     | no
| Fixed issues | #994

#### Summary
Remove all obsolete service message system commands and instead try to execute any user-defined overrides directly in `GenericmessageCommand`.

#### Usage
You need to set `SystemCommand::$execute_deprecated = true;` before handling the request, either in your hook or getUpdates file.

If you have a custom `GenericmessageCommand` implementation, you'll need to add the code snippet from the original `GenericmessageCommand.php` to your own.

### CODE REVIEW

**Comment by jacklul** [comment]
I don't think renaming `GenericmessageCommand` to `MessageCommand` is a good idea, might get confusing when people create `/message` commands or something

Assuming we allow users to use their own namespaces (#919) they should be able simply to extend this and `parent::execute()` instead of creating new class and copying code, this also prevents people from removing essential code for handling conversations.

File: src/Commands/SystemCommands/GenericmessageCommand.php [hunk 1]
```diff
@@ -68,9 +68,57 @@ public function execute()
 
         //Fetch conversation command if it exists and execute it
         if ($conversation->exists() && ($command = $conversation->getCommand())) {
-            return $this->telegram->executeCommand($command);
+            return $this->getTelegram()->executeCommand($command);
+        }
+
+        // Try to execute any deprecated system commands.
+        if ($deprecated_system_command_response = $this->executeDeprecatedSystemCommand()) {
```
Comment (jacklul) [inline src/Commands/SystemCommands/GenericmessageCommand.php]: I think `executeNoDb()` should call this as well?

Comment (noplanman) [reply src/Commands/SystemCommands/GenericmessageCommand.php]: Absolutely!

**Comment by noplanman** [comment]
@jacklul I've updated this a bit and moved the code to `SystemCommand`, to make the integration into a custom `GenericmessageCommand` easier.

Also, there's a property `SystemCommand::$execute_deprecated` which needs to be set for this to work. I added this, to not have all the code run for users who don't need this feature.

Any thoughts are highly appreciated! 🙏

I'm also thinking about how best to implement the new command system, to make it logical and easy to use.

**Pull request merged**